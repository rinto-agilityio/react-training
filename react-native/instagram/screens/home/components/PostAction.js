import React from "react";
import {
  View,
  Text,
  Image,
  TouchableHighlight,
  StyleSheet
} from "react-native";

export default class PostAction extends React.Component {
  hanleLike() {
    const { postId, userId, toogleLike } = this.props;

    toogleLike({
      postId,
      userId
    });
  }

  render() {
    return (
      <View style={styles.wrapper}>
        <TouchableHighlight onPress={() => this.hanleLike()}>
          <Image
            style={styles.icon}
            source={require("../../../assets/icons/like.png")}
          />
        </TouchableHighlight>
        <Image
          style={styles.icon}
          source={require("../../../assets/icons/comment.png")}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    flexDirection: "row"
  },
  icon: {
    width: 20,
    height: 20,
    marginRight: 10
  }
});
