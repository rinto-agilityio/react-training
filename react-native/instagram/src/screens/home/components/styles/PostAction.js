import { StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    flexDirection: "row"
  },
  icon: {
    width: 20,
    height: 20,
    marginRight: 10
  }
});
