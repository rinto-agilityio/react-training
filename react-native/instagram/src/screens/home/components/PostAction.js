// Third party libs
import React from 'react'
import { View, Text, Image, TouchableHighlight } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import { styles } from './styles/PostActionStyles'
import CommonStyles from '@themes/common'

const PostAction = ({
  postId,
  userId,
  isLiked,
  toogleLike,
  toogleCommentBox
}) => {
  // User like or remove liked a photo
  const _handleClickIconLike = () => {
    toogleLike({
      postId,
      userId
    })
  }

  // Toggle comment input on single photo
  const _handleClickIconComment = () => {
    toogleCommentBox()
  }

  const likeIcon = isLiked
    ? require('@assets/icons/liked.png')
    : require('@assets/icons/like.png')

  return (
    <View style={CommonStyles.layoutRow}>
      <TouchableHighlight
        className="icon-like"
        onPress={() => _handleClickIconLike()}
        underlayColor={'transparent'}
      >
        <Image style={styles.icon} source={likeIcon} />
      </TouchableHighlight>
      <TouchableHighlight
        className="icon-comment"
        onPress={() => _handleClickIconComment()}
        underlayColor={'transparent'}
      >
        <Image
          style={styles.icon}
          source={require('@assets/icons/comment.png')}
        />
      </TouchableHighlight>
    </View>
  )
}

PostAction.defaultProps = {
  postId: 0,
  userId: 0,
  isLiked: false
}

PostAction.propTypes = {
  postId: PropTypes.number.isRequired,
  userId: PropTypes.number.isRequired,
  isLiked: PropTypes.bool.isRequired,
  toogleLike: PropTypes.func,
  toogleCommentBox: PropTypes.func
}

export default PostAction
