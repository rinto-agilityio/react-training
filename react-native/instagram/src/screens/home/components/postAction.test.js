import PostAction from './PostAction';

describe('PostAction component', () => {
  let imageUri, iconLike, iconComment;
  const mockToogleComment = jest.fn(),
    mockToogleLike = jest.fn();

  beforeEach(() => {
    const component = shallow(
      <PostAction
        toogleLike={mockToogleLike}
        toogleComment={mockToogleComment}
      />
    );

    iconLike = component.find('TouchableHighlight.icon-like');
    iconComment = component.find('TouchableHighlight.icon-comment');
  });

  it('Renders correctly for new image', () => {
    const treeDOM = renderer.create(<PostAction liked={false} />).toJSON();

    expect(treeDOM).toMatchSnapshot();
  });

  it('Renders correctly for like image', () => {
    const treeDOM = renderer.create(<PostAction liked={true} />).toJSON();

    expect(treeDOM).toMatchSnapshot();
  });

  it('Should call toogleLike if click on icon like', () => {
    iconLike.simulate('press');

    expect(mockToogleLike.mock.calls.length).toEqual(1);
  });

  it('Should call toogleComment if click on icon like', () => {
    iconComment.simulate('press');

    expect(mockToogleComment.mock.calls.length).toEqual(1);
  });
});
