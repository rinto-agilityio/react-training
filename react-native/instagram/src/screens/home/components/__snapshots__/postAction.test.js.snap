// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostAction component Renders correctly for like image 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/liked.png",
        }
      }
      style={
        Object {
          "height": 24,
          "marginRight": 10,
          "width": 24,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/comment.png",
        }
      }
      style={
        Object {
          "height": 24,
          "marginRight": 10,
          "width": 24,
        }
      }
    />
  </View>
</View>
`;

exports[`PostAction component Renders correctly for new image 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/like.png",
        }
      }
      style={
        Object {
          "height": 24,
          "marginRight": 10,
          "width": 24,
        }
      }
    />
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/comment.png",
        }
      }
      style={
        Object {
          "height": 24,
          "marginRight": 10,
          "width": 24,
        }
      }
    />
  </View>
</View>
`;
